app/src/main/java/com/zero/visitnepal/ui/home/HomeFragment.kt

```diff
@@ -25,9 +38,45 @@ class HomeFragment : Fragment() {
         val binding: FragmentHomeBinding =
             DataBindingUtil.inflate(inflater, R.layout.fragment_home, container, false)
 
-        binding.pager.adapter = HomeViewPagerAdapter()
+        (activity?.applicationContext as App).appComponent.inject(this)
+
+        binding.pager.adapter =
+            HomeViewPagerAdapter()
         binding.indicator.setViewPager(binding.pager)
         binding.pager.setPageTransformer(ZoomOutPageTransformer())
+
+        //city adapter
+        homeCityAdapter = HomePlacesAdapter()
+        binding.homeCityListRv.adapter = homeCityAdapter
+
+        //Attraction adapter
+        homeAttractionAdapter = HomePlacesAdapter()
+        binding.homeTopAttractionListRv.adapter = homeAttractionAdapter
```

Comment:
xml id are all lower case , for example, here , should be hometopattractionlistrv , also could rename to
 homeattractionlistrv for consistency